roc.go

```diff
@@ -0,0 +1,136 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat
+
+import "sort"
+
+// ROC returns paired false positive rate (FPR) and true
+// positive rate (TPR) values corresponding to n
+// cutoffs spanning the relative (or receiver)
+// operator characteristic (ROC) curve obtained when y is
+// treated as a binary classifier for classes with weights.
+// Cutoffs are equally spaced from eps less than the minimum value of y
+// to the maximum value of y, including both endpoints -- this means the
+// resulting ROC curve will always begin at (0,0) and end at (1,1).
+// input y must be sorted, and SortWeightedLabelled can be used
+// in order to sort data y together with classes and weights.
+// For a given cutoff value, observations corresponding to
+// entries in y greater than the cutoff are classified as
+// false, and these assigned class labels are compared to
+// classes to calculate FPR and TPR.
+// weights being nil results in the unweighted case.
+// n of zero results in all possible cutoffs being calculated.
```

Comment:
Yep, also that calculating len(y)+1 equally spaced cutoffs is not the same as calculating the len(y)+1 unique (FPR,TPR) pairs. So thats why I made that the behavior when n==0, because essentially this is `special' -- in this case, the cutoffs are _not_ equally spaced in y, they are equally spaced in number of observations if that makes sense.